source/Octopus.Tentacle.Kubernetes.Tests.Integration/KubernetesAgent/KubernetesAgentMigrateFromPreinstallationTest.cs

```diff
@@ -0,0 +1,189 @@
+using System.Text;
+using FluentAssertions;
+using k8s;
+using k8s.Models;
+using Octopus.Diagnostics;
+using Octopus.Tentacle.Commands;
+using Octopus.Tentacle.Diagnostics;
+using Octopus.Tentacle.Startup;
+
+namespace Octopus.Tentacle.Kubernetes.Tests.Integration.KubernetesAgent;
+
+public class KubernetesAgentMigrateFromPreinstallationTest
+{
+    
+    readonly ISystemLog systemLog = new SystemLog();
+    
+    [Test]
+    public async Task MigrationFromPreinstallHookShouldCopyData()
+    {
+        //Arrange
+        var kubernetesConfigClient = new KubernetesFileWrappedProvider(KubernetesTestsGlobalContext.Instance.KubeConfigPath);
+        var commandNamespace = Guid.NewGuid().ToString("N");
+        Environment.SetEnvironmentVariable(KubernetesConfig.NamespaceVariableName, commandNamespace);
+        var commandToRun = new MigratePreInstalledKubernetesDeploymentTargetCommand(new Lazy<IKubernetesClientConfigProvider>(kubernetesConfigClient), systemLog, new LogFileOnlyLogger());
+        var client = new k8s.Kubernetes(kubernetesConfigClient.Get());
+        var validationKey = Guid.NewGuid().ToString("N");
```

Comment:
It would be good to pull out the stuff that isn't related to what you're testing/doesn't change into a `Setup` method or just a `GetCommand` function + move stuff like the name consts out of each test since they are just there to make things work.